src/java/org/apache/cassandra/utils/memory/BufferPool.java

```diff
@@ -214,6 +216,50 @@ public ByteBuffer getAtLeast(int size, BufferType bufferType)
             return localPool.get().getAtLeast(size);
     }
 
+
+    /**
+     * Allocate the given amount of memory, where the caller can accept the space to be split into multiple buffers.
+     *
+     * @param totalSize the total size to be allocated
+     * @param chunkSize the minimum size of each buffer returned
+     *
+     * @return an array of allocated buffers
+     */
+    public ByteBuffer[] getMultiple(int totalSize, int chunkSize, BufferType bufferType)
+    {
+        if (bufferType == BufferType.ON_HEAP || totalSize > NORMAL_CHUNK_SIZE || totalSize <= chunkSize)
+            return new ByteBuffer[] { allocate(totalSize, bufferType) };
+
+        // Try to find a buffer to fit the full request. Fragmentation can make this impossible even if we are below
+        // the limits.
+        LocalPool pool = localPool.get();
+        if (!DISABLE_COMBINED_ALLOCATION)
+        {
+            ByteBuffer full = pool.tryGet(totalSize, false);
+            if (full != null)
+                return new ByteBuffer[]{ full };
+        }
+
+        // If we don't get a whole buffer, allocate buffers of the requested minimum size.
+        int numBuffers = (totalSize + chunkSize - 1) / chunkSize;
+        ByteBuffer[] buffers = new ByteBuffer[numBuffers];
+        int remainingSize = totalSize;
+        int idx = 0;
+
+        while (remainingSize >= chunkSize)
+        {
+            buffers[idx] = pool.getAtLeast(chunkSize);
```

Comment:
For my own curiosity, what is the rational for when to use `getAtLeast` versus `get`? It seems that `getAtLeast` still return a buffer whose capacity has been set to `chunkSize` (or does it not?) and if so, why do we use `get` instead of `getAtLeast` in the last call below?

Reply:
Simplified now. I was originally thinking of letting it combine some of these buffers, which wouldn't be that helpful and can't (easily) work with memory addresses.